server/embedded/src/org/labkey/embedded/LabKeyServer.java

```diff
@@ -225,16 +229,23 @@ protected TomcatWebServer getTomcatWebServer(Tomcat tomcat)
                    configureJsonAccessLogging(tomcat, logConfig);
                 }
 
-                WebappProperties additionalWebapps = additionalWebappSource();
-                if (additionalWebapps.getContextPath().size() != additionalWebapps.getDocBase().size())
-                {
-                    throw new IllegalArgumentException("Additional webapps must have paired contextPath and docBase properties");
-                }
-                for (int i = 0; i < additionalWebapps.getContextPath().size(); i++)
+                Map<String, String> additionalWebapps = contextProperties.getAdditionalWebapps();
+                if (additionalWebapps != null)
                 {
-                    String contextPath = additionalWebapps.getContextPath().get(i);
-                    String docBase = additionalWebapps.getDocBase().get(i);
-                    tomcat.addWebapp(contextPath, docBase);
+                    for (Map.Entry<String, String> entry : additionalWebapps.entrySet())
+                    {
+                        String contextPath = entry.getKey();
+                        if (!contextPath.startsWith("/"))
+                        {
+                            contextPath = "/" + contextPath;
+                        }
+                        String docBase = entry.getValue();
+                        if (docBase == null || docBase.isEmpty())
+                        {
+                            throw new IllegalArgumentException("No docBase supplied additional webapp at context path " + contextPath);
```

Comment:
Should this be a `ConfigException`?

Reply:
It can now that I switched `ConfigException` to a `RuntimeException`